### PR TITLE
Support julia 0.5

### LINKED ISSUE #118: Migrate to Julia 0.5-dev
That is, add "nightly" to .travis.yml without causing tests to fail.

### PR DESCRIPTION
**Pull request by jeff-regier**
This PR turns on travis testing for Julia nightly, and makes the tests pass. It isn't always pretty:
- `runtests.jl` checks out `Optim.jl` master and `DataFrames.jl` master
- I forked `GaussianProcesses.jl` and fixed just enough of it. `runtests.jl` clones the fork.
- `JLD.jl` doesn't work yet, though it may soon: JuliaLang/JLD.jl#57  This doesn't stop the tests because the JLD stuff is all in bin.

Still, I'd say the PR closes #118

### CODE REVIEW

File: src/ModelInit.jl [hunk 1]
```diff
@@ -464,7 +464,7 @@ function limit_to_object_data(
   # Get the sources that overlap with this object.
   relevant_sources = Int[]
   for b = 1:length(blob), tile_sources in mp.tile_sources[b]
-    if s_original in tile_sources > 0
```
Comment (jeff-regier) [inline src/ModelInit.jl:467]: We probably didn't mean to compare a boolean to an int before, but for whatever reason 0.4 doesn't complain, while 0.5 stops.

File: src/elbo_kl.jl [hunk 2]
```diff
@@ -72,9 +72,10 @@ function subtract_kl!{NumType <: Number}(
     mp::ModelParams{NumType}, accum::SensitiveFloat{CanonicalParams, NumType};
     calculate_derivs::Bool=true, calculate_hessian::Bool=true)
 
+
   # The KL divergence as a function of the active source variational parameters.
-  function subtract_kl_value_wrapper{NumType <: Number}(vp_vec::Vector{NumType})
-    elbo_val = zero(NumType)
+  function subtract_kl_value_wrapper{NumType2 <: Number}(vp_vec::Vector{NumType2})
```
Comment (jeff-regier) [inline src/elbo_kl.jl:77]: This was causing a seg fault in 0.5, rather than an error

File: test/test_derivatives.jl [hunk 3]
```diff
@@ -16,6 +16,9 @@ else
     end
 end
 
+const datadir = joinpath(Pkg.dir("Celeste"), "test", "data")
```
Comment (jeff-regier) [inline test/test_derivatives.jl]: I might be redefining a constant here, but it's nice to run just one file or tests without having to run all the tests. (Is there a better way?)

Comment (kbarbary) [inline test/test_derivatives.jl]: Yes, just remove this. `datadir` is already defined in `runtests.jl`. All the `test_*.jl` files are included into that file, so they'll be in the same namespace/scope as `datadir` already: including just a subset of the `test_*.jl` files doesn't affect this.

File: test/test_images.jl [hunk 4]
```diff
@@ -4,12 +4,14 @@ using DataFrames
...
-import Celeste: ModelInit, SkyImages, ElboDeriv
+import Celeste: ModelInit, SkyImages, ElboDeriv, Synthetic
 
 
 println("Running SkyImages tests.")
 
-field_dir = joinpath(dat_dir, "sample_field")
+const datadir = joinpath(Pkg.dir("Celeste"), "test", "data")
+
+field_dir = joinpath(datadir, "sample_field")
```
Comment (kbarbary) [inline test/test_images.jl]: There's no `sample_field` subdirectory in `test/data` anymore; the files are all in `test/data`.

File: test/test_transforms.jl [hunk 5]
```diff
@@ -5,7 +5,7 @@ using Base.Test
 
 using Celeste: Types, SampleData, Transform, SensitiveFloats
 import Celeste: ModelInit, ElboDeriv
-
+using Compat
```
Comment (kbarbary) [inline test/test_transforms.jl:8]: Sorry for  removing all the `using Compat`s in #134! I figured most were for 0.3->0.4 changes and thus no longer needed.

File: test/runtests.jl [hunk 6]
```diff
@@ -1,5 +1,12 @@
 #!/usr/bin/env julia
 
+if VERSION > v"0.5.0-dev"
+    Pkg.checkout("Optim")
+    Pkg.checkout("DataFrames")
+    Pkg.rm("GaussianMixtures")
+    Pkg.clone("https://github.com/jeff-regier/GaussianMixtures.jl.git")
```
Comment (kbarbary) [inline test/runtests.jl]: These run-time changes to which packages are installed seem like a bad idea. Better to document in the README the requirements for running on v0.5.

Comment (kbarbary) [inline test/runtests.jl]: (and for travis, add these commands to `.travis.yml`)

File: test/test_derivatives.jl (hunk 3 above)
Comment (jeff-regier) [inline test/test_derivatives.jl]: Oh I see, I can call runtests with ARGS now.

File: test/test_images.jl (hunk 4 above)
Comment (jeff-regier) [inline test/test_images.jl]: It's weird we have both `dat_dir` defined in SampleData.jl and `datadir` defined `runtests.jl`. SampleData.jl probably shouldn't know about the test data's location.

File: test/test_transforms.jl (hunk 5 above)
Comment (jeff-regier) [inline test/test_transforms.jl:8]: Removing them before seemed like a good thing to do...I didn't realize how soon we'd need them again

File: test/test_images.jl (hunk 4 above)
Comment (kbarbary) [inline test/test_images.jl]: > It's weird we have both dat_dir defined in SampleData.jl and datadir
> defined runtests.jl. SampleData.jl probably shouldn't know about the test
> data's location.

I agree. Probably SampleData.jl should eventually be moved to tests, as its
not necessary for the core functionality in the app. This is something to
think about after SC though!

**Pull request merged**